var/spack/repos/builtin/packages/petsc/package.py

```diff
@@ -113,11 +113,14 @@ def mpi_dependent_options(self):
                 raise RuntimeError('\n'.join(errors))
         else:
             compiler_opts = [
-                '--with-cpp=cpp',
-                '--with-cxxcpp=cpp',
                 '--with-mpi=1',
                 '--with-mpi-dir=%s' % self.spec['mpi'].prefix,
             ]
+        if sys.platform != "darwin":
```

Comment:
btw, if this is Sierra only + `gcc` isse, the `if` should reflect this fact. It certainly builds fine with `clang` on Sierra! See https://github.com/LLNL/spack/issues/1814. For now I would at least suggest combining 
```
mac_ver = '.'.join(py_platform.mac_ver()[0].split('.')[:2])
```
with 
```
if spec.satisfies('%gcc')
```
Turning this off for all builds on `darwin` does not look right to me.

@tgamblin is there a variable in Spack which holds version of OS to be used in conditionals inside `package.py`?